### DISCUSSION TITLE
SMA4 findings

### DISCUSSION THREAD

**Discussion by nintyalex** [city41/smaghetti#62] [General]
I've found out that the flagpole's length (bank 04, id 7f) can be altered. The second digit of the first byte controls its height (excluding the block it rests on). The flag is a separate sprite (bank 01, id 1c); it determines where the top and touchable pole are.

The blank tile sprites you mentioned control horizontal scrolling. Sprite bank 00, id c7 makes the camera stop scrolling on the right side, and bank 00, id ea does the same for the left side. The last byte says how many pixels to stop before that point. It's how Nintendo fit the original airships' boss rooms onto the screen.

**Comment by city41** [comment]
Oh yeah, I forgot to document that about flag poles. I chose the height I did as it matches SMB1 flag poles. I'll make an issue now to add a config to flag poles to let people choose their height.

oh nice, that makes sense about the scrolling sprites. ok, cool. I would imagine smaghetti won't get involved in stuff like that for a while.

**Comment by city41** [comment]
I think for those "meta" sprites, I'll somehow add them to hex-tree so they stop showing up as unknown.

**Reply by city41** [reply]
btw @nintyalex if you want to be notified on new discussions, you can choose to watch the repo, then set the watch to custom and pick discussions.
